hedera-mirror-rest/tokens.js

```diff
@@ -400,42 +398,40 @@ const extractSqlFromTokenInfoRequest = (tokenId, filters) => {
   const conditions = [`${CustomFee.TOKEN_ID} = $1`];
   const params = [tokenId];
 
+  let customFeeQuery = `
+        ${customFeeSelect}
+        from ${CustomFee.tableName}
+        where ${conditions.join(' and ')}`;
   if (filters && filters.length !== 0) {
     // honor the last timestamp filter
     const filter = filters[filters.length - 1];
     const op = transformTimestampFilterOp(filter.operator);
-    conditions.push(`${CustomFee.FILTER_MAP[filter.key]} ${op} $2`);
+    conditions.push(`lower(${CustomFee.TIMESTAMP_RANGE}) ${op} $2`);
     params.push(filter.value);
-  }
 
-  const aggregateCustomFeeQuery = `
-    select jsonb_agg(jsonb_build_object(
-                       'all_collectors_are_exempt', ${CustomFee.ALL_COLLECTORS_ARE_EXEMPT},
-                       'amount', ${CustomFee.AMOUNT},
-                       'amount_denominator', ${CustomFee.AMOUNT_DENOMINATOR},
-                       'collector_account_id', ${CustomFee.COLLECTOR_ACCOUNT_ID}::text,
-                       'created_timestamp', ${CustomFee.CREATED_TIMESTAMP}::text,
-                       'denominating_token_id', ${CustomFee.DENOMINATING_TOKEN_ID}::text,
-                       'maximum_amount', ${CustomFee.MAXIMUM_AMOUNT},
-                       'minimum_amount', ${CustomFee.MINIMUM_AMOUNT},
-                       'net_of_transfers', ${CustomFee.NET_OF_TRANSFERS},
-                       'royalty_denominator', ${CustomFee.ROYALTY_DENOMINATOR},
-                       'royalty_numerator', ${CustomFee.ROYALTY_NUMERATOR},
-                       'token_id', ${CustomFee.TOKEN_ID}::text
-                       )
-                     order by ${CustomFee.COLLECTOR_ACCOUNT_ID}, ${CustomFee.DENOMINATING_TOKEN_ID}, ${
-    CustomFee.AMOUNT
-  }, ${CustomFee.ROYALTY_NUMERATOR})
-    from ${CustomFee.tableName} ${CustomFee.tableAlias}
-    where ${conditions.join(' and ')}
-    group by ${CustomFee.getFullName(CustomFee.CREATED_TIMESTAMP)}
-    order by ${CustomFee.getFullName(CustomFee.CREATED_TIMESTAMP)} desc
-    limit 1
-  `;
+    // include the history table in the query
+    customFeeQuery = `
+        with cfee as (
+          ${customFeeSelect}
+          from ${CustomFee.tableName}
+          where ${conditions.join(' and ')}
+        ),
+        custom_fee_history as (
+          ${customFeeSelect}  
+          from ${CustomFee.tableName}_history
+          where ${conditions.join(' and ')}
+          order by lower(${CustomFee.TIMESTAMP_RANGE}) desc
+          limit 1
+        )
+        select * from cfee
+        union all
+        select * from custom_fee_history
+        limit 1`;
```

Comment:
the union all may produce two rows, without `order by`, it'll be random which row we get with the `limit 1`

```suggestion
        select * from cfee
        union all
        select * from custom_fee_history
        order by lower(${CustomFee.TIMESTAMP_RANGE}) desc
        limit 1`;
```

Reply:
This has now been updated.